sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/postgresql.yaml

```diff
@@ -27,10 +27,27 @@ provider:
           # Certain special characters need to be percent-encoded so we can break matching when we find them.
           pattern: "\\bpostg(?:res(?:ql)?|is)(?:\\+\\w+)?://[^:@/ ]+:([^@/ ]+)@"
         post:
-          # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
-          # Note: Anything matching values like "${DB_PASSWORD:-postgres}" will be ignored and processed by another
-          # rule.
-          patternNot: "(?i)^(\\$[a-z_]*)?(db|my)?_?pass(word|wd)?|^\\[[\\w\\t \\-]+\\]$|^<[\\w\\t \\-]+>$|^%s$|^\\{+[^}]*\\}+$|^\\$\\(\\w+\\)$|\\$\\{\\w+(:-\\w+)?\\}"
+          # Potential FPs here are environment variables, templates and string substitutions, such as
+          # - $my_password
+          # - $db_password
+          # - [mypass123]
+          # - <example-password>
+          # - %s
+          # - {{MyPassword}}
+          # - $(mypassword)
+          # - ${mypassword}
+          # - os.getenv("PASS")
+          # - os.environ['PASS']
+          patternNot:
+            "(?i)\
+            ^(\\$[a-z_]*)?(db|my)?_?pass(word|wd)?|\
+            ^\\[[\\w\\t \\-]+\\]$|\
+            ^<[\\w\\t \\-]+>$|\
+            ^%s$|\
+            ^\\{+[^}]*\\}+$|\
+            ^\\$\\(\\w+\\)$|\
+            \\$\\{\\w+(:-\\w+)?\\}|\
+            \\b(get)?env(iron)?\\b"
```

Comment:
I suggest removing as much patternNot as possible to try to clean up as much as we can, and adding that to the provider detection block instead:

```
  detection:
    post:
      # Potential FPs here are environment variables, templates and string substitutions, such as
      # - $my_password
      # - $db_password
      # - [mypass123]
      # - <example-password>
      # - %s
      # - {{MyPassword}}
      # - $(mypassword)
      # - ${mypassword}
      # - os.getenv("PASS")
      # - os.environ['PASS']
      # - `myPassword`
      # - *********
      # - None
      patternNot:
        "(?i)\
        ^(\\$[a-z_]*)?(db|my)?_?pass(word|wd)?|\
        ^\\[[\\w\\t \\-]+\\]$|\
        <[\\w\\t \\-]+>?|\
        ^%s$|\
        ^\\{+[^}]*\\}+$|\
        ^\\$\\(\\w+\\)$|\
        \\$\\{\\w+(:-\\w+)?\\}|\
        ^\\$[{(]|\
        \\b(get)?env(iron)?\\b|\
        ^`[a-z]+|\
        ^\\*{3,}$|\
        ^None$"
```